vm/src/stdlib/pyexpat.rs

```diff
@@ -0,0 +1,204 @@
+        }
+
+        fn create_config(&self) -> xml::ParserConfig {
+            xml::ParserConfig::new()
+                .cdata_to_characters(true)
+                .coalesce_characters(false)
+                .whitespace_to_characters(true)
+        }
+
+        fn do_parse<T>(&self, vm: &VirtualMachine, parser: xml::EventReader<T>)
+        where
+            T: std::io::Read,
+        {
+            for e in parser {
+                match e {
+                    Ok(XmlEvent::StartElement {
+                        name, attributes, ..
+                    }) => {
+                        let dict = vm.ctx.new_dict();
+                        for attribute in attributes {
+                            dict.set_item(
+                                attribute.name.local_name.as_str(),
+                                vm.ctx.new_str(attribute.value),
+                                vm,
+                            )
+                            .unwrap();
+                        }
+
+                        let name_str = PyStr::from(name.local_name).into_ref(vm);
+                        invoke_handler(vm, &self.start_element, (name_str, dict.into_object()));
+                    }
+                    Ok(XmlEvent::EndElement { name, .. }) => {
+                        let name_str = PyStr::from(name.local_name).into_ref(vm);
+                        invoke_handler(vm, &self.end_element, (name_str,));
+                    }
+                    Ok(XmlEvent::Characters(chars)) => {
+                        let str = PyStr::from(chars).into_ref(vm);
+                        invoke_handler(vm, &self.character_data, (str,));
+                    }
+                    _ => {}
+                }
+            }
+        }
+
+        #[pymethod(name = "Parse")]
+        fn parse(&self, data: PyStrRef, isfinal: OptionalArg<bool>, vm: &VirtualMachine) {
+            let reader = Cursor::<Vec<u8>>::new(data.as_str().as_bytes().iter().cloned().collect());
+            let parser = self.create_config().create_reader(reader);
+            self.do_parse(vm, parser);
+        }
+
+        #[pymethod(name = "ParseFile")]
+        fn parse_file(&self, file: PyObjectRef, vm: &VirtualMachine) -> PyResult<()> {
+            // todo: read chunks at a time
+            let read_res = vm.call_method(&file, "read", ())?;
+            let bytes_like = PyBytesLike::try_from_object(vm, read_res)?;
+            let buf = bytes_like.borrow_buf().to_vec();
+            let reader = Cursor::new(buf);
+            let parser = self.create_config().create_reader(reader);
+            self.do_parse(vm, parser);
+
+            // todo: return value
+            Ok(())
+        }
+    }
+
+    #[derive(FromArgs)]
+    struct ParserCreateArgs {
+        #[pyarg(any, optional)]
+        encoding: OptionalArg<PyStrRef>,
+        #[pyarg(any, optional)]
+        namespace_separator: OptionalArg<PyStrRef>,
+        #[pyarg(any, optional)]
+        intern: OptionalArg<PyStrRef>,
```

Comment:
What's `intern`s intended use? I can't see it in the signature of `ParserCreate` in the docs.

Reply:
I saw it in CPython source. I can remove it though, we're not using this argument now.